chore(lints): introduce typos-cli config in `Cargo.toml` and handle typos

mkatychev opened the pull request:
related PR: https://github.com/tweag/topiary/pull/832

* Added typos config to `Cargo.toml`
  https://github.com/tweag/nickel/blob/d77b6c35ec2db5200a7f343bdf21ec6d5a0e1576/Cargo.toml#L147-L157
* Ran [`typos -w`](https://github.com/crate-ci/typos) on the project
* Manually fixed ambiguous resolution typos: https://github.com/tweag/nickel/commit/d77b6c35ec2db5200a7f343bdf21ec6d5a0e1576

mkatychev commented:
Not sure where this line/column diff is coming from 🤔 https://github.com/tweag/nickel/actions/runs/18698880001/job/53323091230?pr=2390#step:5:1848

jneem commented:
It may be best to redo this after #2302 is finished -- otherwise there will be a lot of conflicts

mkatychev commented:
Sounds good, all the manual steps have been documented so I can just rerun all of it.

mkatychev committed:
added typos config

mkatychev committed:
added more whitelist

mkatychev committed:
manual fix

mkatychev committed:
typos --no-check-filenames --write-changes

mkatychev committed:
manual fixes:
sd 'seach' 'search' ./**/*.rs
sd 'seach' 'search' ...(fd -e rs)
sd 'dyamic' 'dynamic' ./**/*.md
sd 'dyamic' 'dynamic' ...(fd)
sd 'wich' 'which' ...(fd)
sd 'contining' 'containing' ./**/*.rs
sd 'contining' 'containing' ...(fd)
sd 'contine' 'continue' ...(fd)
sd 'evalution' 'evaluation' ...(fd)
typos --no-check-filenames
sd 'overrided' 'overridden' ...(fd)
^sd 'doens' 'doesn' ./**/*.rs
^sd 'shrinked' 'shrunk' ./**/*.rs
sd 'represention' 'representation' ...(fd)
^sd 'confing' 'config' ./**/*.md
^sd 'chaning' 'changing' ./**/*.rs

mkatychev committed:
fix filenames

mkatychev committed:
fix snapshot

mkatychev committed:
fixup false negative

jneem reviewed (approved):
👍
